website/project/views/contributor.py

```diff
@@ -509,7 +522,16 @@ def notify_added_contributor(node, contributor, auth=None, throttle=None, email_
     if (contributor.is_registered and not node.template_node and not node.is_fork and
             (not node.parent_node or
                 (node.parent_node and not node.parent_node.is_contributor(contributor)))):
-        email_template = getattr(mails, 'CONTRIBUTOR_ADDED_{}'.format(email_template.upper()))
+
+        preprint_provider = None
+        if email_template == 'preprint':
```

Comment:
(Add an author who has an OSF account) Query the database and find if the project is associated with a preprint. If so, use the customized templates.